### PR TITLE
Refactor and Add Task O Tutorial

### PR DESCRIPTION
**Pull request by KyleKing**
I'm trying to think through what to use for Task 2 and refactoring as I figure out what can be extracted. This is just a WIP PR that could be merged early, but I'll keep it open until I have something workable

### CODE REVIEW

File: game/tasks/task03_l_crafting.py [hunk 1]
```diff
@@ -10,16 +10,21 @@
...
 An important step in designing reusable code is to find opportunities to write code that
 is interchangeable.
 
-# TODO: I think the logic to build items would be a good use case where the class needs to handle the interface for different items that output a new sprite
-
 """
 
-SOURCE_NAME = "task03_L_TBD"  # FYI: Required for code reload
+
+# TODO: I think the Inventory could be a good example where we provide a naive implementation that has all of the logic implemented together, but then we can demo how one aspect could be split, then leave it up to the user to make the rest of the refactoring
```
Comment (KyleKing) [inline game/tasks/task03_l_crafting.py:22]:
```suggestion
# TODO: I think the logic to build items would be a good use case where the class needs to handle the interface for different items that output a new sprite
```

**Review by KyleKing — COMMENTED** [review]
I'm stuck on figuring out what the player should change for the OCP example. We don't currently have any code that seems like a good candidate for extraction for this task. A lot of the code that could be useful touches multiple internal classes that would be confusing for the player

Can you think of any code that might make for a good standalone example?

**Review by dimmerex — APPROVED** [review]
Approved! O task looks good so far, I think you selected a good example to teach. Only thought is maybe we should keep code blocks commented out if the user isn't expected to modify it?

**Comment by KyleKing** [comment]
Hm, having the code I commented means that it will have syntax highlighting and they can run it, but it would be good to clarify what needs to be edited.

We could probably add "Part 1 (Lesson)" in the divider?l vs. "Part 2 (Task)"?

**Pull request merged**